plutus-tx/src/PlutusTx/Show.hs

```diff
@@ -20,28 +21,78 @@ import PlutusTx.Bool
 import PlutusTx.Builtins qualified as Builtins
 import PlutusTx.Either
 import PlutusTx.Foldable (foldl)
+import PlutusTx.List
 import PlutusTx.Maybe
 import PlutusTx.Show.TH
 
 instance Show Builtins.Integer where
-    {-# INLINEABLE show #-}
-    show = Builtins.showInteger
+    {-# INLINEABLE showsPrec #-}
+    showsPrec _ n =
+        showString (if n `Builtins.lessThanInteger` 0 then "-" else "")
+            . foldr alg id (toDigits n)
+      where
+        toDigits :: Builtins.Integer -> [Builtins.Integer]
+        toDigits m =
+            let quotient = m `Builtins.quotientInteger` 10
+                remainder = m `Builtins.remainderInteger` 10
+                digit =
+                    if remainder `Builtins.lessThanInteger` 0
+                        then 0 `Builtins.subtractInteger` remainder
+                        else remainder
+             in if quotient `Builtins.equalsInteger` 0
+                    then [digit]
+                    else digit : toDigits quotient
```

Comment:
Can we have a property test that generates thousands of integers, shows each of them via this function, reads back the result via regular Haskell `Read` and checks that it's the same number?

Reply:
Sounds good, will do.

Reply:
I also think just defining the inverses ourselves is maybe a good idea. They might be useful? For testing at least.